extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -310,8 +312,8 @@ public Table listSnapshotsAsTable(@NotNull final TableIdentifier tableIdentifier
         columnSourceMap.put("Id", InMemoryColumnSource.getImmutableMemoryColumnSource(idArr, long.class, null));
 
         final long[] timestampArr = new long[(int) size];
-        columnSourceMap.put("TimestampMs",
-                InMemoryColumnSource.getImmutableMemoryColumnSource(timestampArr, long.class, null));
+        columnSourceMap.put("Timestamp", ReinterpretUtils.longToInstantSource(
```

Comment:
This is a bit backwards. Make the `Instant` source initially, and then reinterpret it to long for filling purposes. The factories will even let you use a `long` array.